posthog/api/signup.py

```diff
@@ -0,0 +1,236 @@
+from typing import Any, Dict, Optional, cast
```

Comment:
Copied over from `organization.py` instead of minor changes noted within the file and dropping "Organization" from all class names, e.g. `OrganizationSocialSignupViewset` becomes `SocialSignupViewset`.